app/services/charges/charge_models/percentage_service.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Charges
+  module ChargeModels
+    class PercentageService < Charges::ChargeModels::BaseService
+      protected
+
+      def compute_amount(value)
+        amount = value * BigDecimal(rate)
+
+        return amount if BigDecimal(fixed_amount_value).zero?
```

Comment:
Since the methods calls 3 times `BigDecimal(fixed_amount_value)` what do you think of adding a method for it?

Reply:
Just a suggestion, but I feel it could be a bit clearer with something like that:

```ruby
def compute_amount(value)
  compute_percent_amount(value) + compute_fixed_amount(value)
end

def compute_rate_amount(value)
  value * rate
end

def compute_fixed_amount(value)
  return fixed_amount if fixed_amount_target == 'all_units'
  
  value * fixed_amount_value
end

def rate
  @rate ||= BigDecimal(charge.properties['rate'])
end

def fixed_amount
  @fixed_amount ||= BigDecimal(charge.properties['fixed_amount_value'] || 0)
end
```